src/java.base/share/classes/sun/security/util/DerValue.java

```diff
@@ -956,13 +1145,39 @@ public void resetTag(byte tag) {
         this.tag = tag;
     }
 
+    /**
+     * Returns a new DerValuw with a different tag.
+     *
+     * @param newTag the new tag
+     * @return a new DerValue
+     */
+    public DerValue withTag(byte newTag) {
+        return new DerValue(newTag, buffer, start, end, allowBER);
+    }
+
     /**
      * Returns a hashcode for this DerValue.
      *
      * @return a hashcode for this DerValue.
      */
     @Override
     public int hashCode() {
-        return toString().hashCode();
+        int result = tag;
+        for (int i = start; i < end; i++) {
+            result = 31 * result + buffer[i];
+        }
+        return result;
+    }
+
+    DerValue[] subs(byte expectedTag, int startLen) throws IOException {
+        if (expectedTag != 0 && expectedTag != tag) {
```

Comment:
so if expectedTag == 0, then we don't check for tag match? Looks strange. If this is intentional, we should add comment to clarify why.

Reply:
For IMPLICIT. Will comment.